woocommerce/google-listings-and-ads
# Implement new components for phone verification process

### Changes proposed in this Pull Request:



Part of #962.

- Implement a new component `<VerificationCodeControl>` for inputting verification code
- Implement a new component `<VerifyPhoneNumberContent>` for phone verification process
- Add a temporary demo to the Settings page.

### Additional notes

- [x] 📝  The `<DemoVerifyPhoneNumberCard>` should be removed before merging this PR.
- 💡 This PR added new components for subsequence use. It will not change any implementation of the current lite contact info.

### Screenshots:

#### 🎥 . New components for easily entering verification codes

https://user-images.githubusercontent.com/17420811/131971354-5ddfe145-7158-4ecb-bde1-84f40012b9e7.mp4

- Auto-focus to next `<input>` node after entering a valid digit
- Use the arrow keys to move the input cursor
- When key down on the backspace,
    - remove the current digit if the cursor is at the right of digit.
    - move the previous `<input>` if the cursor is at the beginning. And if there's a digit in the previous `<input>`, it would be removed.
- Replace existed digit with the new inputting one

#### 🎥 . The phone verification process

https://user-images.githubusercontent.com/17420811/131969645-6e768b22-903a-4a74-ab75-456ae73debbb.mp4

- Switch verification method and cooldown of the resend button
- The cooldown countdown of the two methods are independent of each other
- Show error message when the verification is failed
- A callback would be invoked when verification is successful

#### 🎥 . Request verification code again

https://user-images.githubusercontent.com/17420811/131970510-c72ca56e-c1da-4333-a22f-801bcc3a971b.mp4

- The verification code request of the same method would be trigger only once when switching the method
- Use the "Resend code" or "Call again" to request code again, and all `<input>` of verification code would be reset to empty. (They would also be reset when switching method but I forgot to record it in. 😅 )
- \*To facilitate the recording of this demo, I changed the cooldown time to 5 seconds locally.

### Detailed test instructions:

1. Go to the Setting page and there's a demo of the two new components for testing
2. Try the UI to see if they work properly and whether it's intuitive and easy to use by entering the verification code.
3. Enter the verification code `000000` to pass the verification, and the rest will fail.

### Changelog entry
